### PR TITLE
2847 release flow

### PR DESCRIPTION
**Pull request by omerzam**
# Description

modify goodworker to trigger QA/prod release via Github actions workflow_dispatch instead of Travis

About # ([2847](https://app.zenhub.com/workspaces/gooddollar-5d50833888a83c6880d3e345/issues/gooddollar/gooddapp/2847))

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

# Checklist:
- [ ] PR title matches follow: (Feature|Bug|Chore) Task Name
- [X] My code follows the style guidelines of this project
- [X] I have followed all the instructions described in the initial task (check Definitions of Done)
- [X] I have performed a self-review of my own code
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added reference to a related issue in the repository
- [ ] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [ ] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

**Comment by omerzam** [comment]
This code had not been tested since there is no dev env for this service

**Comment by serdiukov-o-nordwhale** [comment]
> This code had been tested since there is no dev env for this service

@omerzam 
maybe you meant "have **not** been tested" ?

File: index.js [hunk 1]
```diff
@@ -3,17 +3,18 @@
...
-const AMPLITUDE_KEY = process.env.AMPLITUDE_KEY
+const BOT_NAME = 'GoodDollar Support'
 const AMPLITUDE_SECRET = process.env.AMPLITUDE_SECRET
 const SENTRY_PROJECT_ID = process.env.SENTRY_PROJECT
 const SENTRY_KEY = process.env.SENTRY_KEY
 const REAMAZE_USER = process.env.REAMAZE_USER
 const REAMAZE_TOKEN = process.env.REAMAZE_TOKEN
-const TRAVIS_TOKEN = process.env.TRAVIS_TOKEN
 const SLACK_MONITORING_WEBHOOK = process.env.SLACK_MONITORING_WEBHOOK
+const GITHUB_DAPP_WORKFLOW_ID = process.env.GITHUB_DAPP_WORKFLOW_ID
```
Comment (serdiukov-o-nordwhale) [inline index.js]: I see we have now a lot of the env vars. Maybe it's time to create config.js file using convict to parse them from dotenv.

File: index.js [hunk 2]
```diff
@@ -135,21 +136,43 @@ const goodserverPost = async (cmd, data, env) => {
...
+      targetbranch: targetBranch,
+    },
   }
-  const response = await fetch('https://api.travis-ci.com/repo/' + repocmd, {
-    method: 'POST', // *GET, POST, PUT, DELETE, etc.
-    headers: {
-      'Content-Type': 'application/json',
-      'Travis-API-Version': 3,
-      Authorization: `token ${TRAVIS_TOKEN}`,
+  const res = await fetch(
```
Comment (serdiukov-o-nordwhale) [inline index.js:151]: a raw request again, maybe it's time to move to the axios ?
- create & configure root API objects
- then just use api.<method>('/relative/endpoint', { data as JS object, will be automatically strginified })

If axios won't work in the worker (i remember you've told about som issues with it) maybe to find some lightweight alternative which allows to create pre-configured api objects

**Review by serdiukov-o-nordwhale — APPROVED** [review]

**Comment by serdiukov-o-nordwhale** [comment]
@omerzam 
about testing - we could do it in 3 steps

1. update worker on prod, replace fetch to console.log. Then test does the worker handled /release command send from slack, does the worker prepares req with the correct endpoint, body, args etc
2. if/when everything looks fine, test request prepared by the worker separately (via curl or any rest api tester).
3. bring fetch back, deploy perform final test. /release command should trigger QA deploy now

Then you could add some option to release prod (e.g. /release prod)
But in that case just test pt 1. for prod (e.g. if the correct request prepares for the prod).
As you already tested 2-3 for QA, 99% it should work also for prod

File: index.js (hunk 1 above)
Comment (omerzam) [reply index.js]: I try to keep the scope of this task as minimal as possible
this service is small isolated and does not change very often

File: index.js (hunk 2 above)
Comment (omerzam) [reply index.js:151]: I am not sure if anything else works in the worker environment
maybe it is possible to wrap it with some kind of method to make it simpler but I think it is over-engineering for this project

**Comment by omerzam** [comment]
> @omerzam
> about testing - we could do it in 3 steps
> 
good idea I can do that
>     1. update worker on prod, replace fetch to console.log. Then test does the worker handled /release command send from slack, does the worker prepares req with the correct endpoint, body, args etc
> 
I already tested the fetch code with a small script on a forked project of DAPP and GoodServer and it does trigger the workflow
>     2. if/when everything looks fine, test request prepared by the worker separately (via curl or any rest api tester).
> 
good idea.
>     3. bring fetch back, deploy perform final test. /release command should trigger QA deploy now
> 
> 
> Then you could add some option to release prod (e.g. /release prod)
> But in that case just test pt 1. for prod (e.g. if the correct request prepares for the prod).
> As you already tested 2-3 for QA, 99% it should work also for prod

the simplest way would be to create a dev env for the worker like me and @sirpy did for the verify worker
@sirpy is it possible to do that easily and quickly?

**Comment by sirpy** [comment]
If you tested on a forked project that's fine. Also simply testing in
preview mode is fine also

**Comment by omerzam** [comment]
> If you tested on a forked project that's fine. Also simply testing in preview mode is fine also

I have tested the code snippet making the request to Github. not the code of the worker as a whole
also to run the worker code locally in preview mode or dev mode I need the values of all the env vars @sirpy

**Comment by omerzam** [comment]
testes worker in dev mode (with all code not related to github commented out) and it works
it should be ready to deploy

**Pull request merged**